### PR TITLE
(PC-17142) feat(home): do not show expired credit for underage

### PR DESCRIPTION
**Pull request by anoukhello**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-17142

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

https://user-images.githubusercontent.com/22886846/192364262-e6593767-e982-4e9c-9466-71a043525a5b.mov

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/home/components/HomeHeader.tsx [hunk 1]
```diff
@@ -16,19 +17,28 @@ import { useCustomSafeInsets } from 'ui/theme/useCustomSafeInsets'
...
-    userInfos?.firstName && isLoggedIn ? `Bonjour ${userInfos.firstName}` : 'Bienvenue\u00a0!'
+    user?.firstName && isLoggedIn ? `Bonjour ${user.firstName}` : 'Bienvenue\u00a0!'
 
-  let subtitle = 'Toute la culture à portée de main'
-  if (userInfos?.isBeneficiary && availableCredit && isLoggedIn) {
-    subtitle = availableCredit.isExpired
-      ? 'Ton crédit est expiré'
-      : `Tu as ${formatToFrenchDecimal(availableCredit.amount)} sur ton pass`
+  // we distinguish 3 types of users here: not connected and eligible-to-credit users,
+  // beneficiary users and ex beneficiary users
```
Comment (yorickeando) [inline src/features/home/components/HomeHeader.tsx:29]:
```suggestion
// we distinguish three different cases: 
// - not connected OR eligible-to-credit users
// - beneficiary users
// - ex-beneficiary users

Comment (yorickeando) [reply src/features/home/components/HomeHeader.tsx:29]: Je trouve ça un peu plus clair, parce qu'ai début je me suis dit "Mais comment peut-on être non connecté et éligible au crédit à la fois ? 🤔"

**Review by yorickeando — APPROVED** [review]
Super cool. un petit commentaire pour ton commentaire

File: src/features/home/components/HomeHeader.tsx (hunk 1 above)
Comment (anoukhello) [reply src/features/home/components/HomeHeader.tsx:29]: J'avoue!

**Pull request merged**